addons/l10n_it_edi/models/account_move_send.py

```diff
@@ -73,11 +73,12 @@ def _call_web_service_after_invoice_pdf_render(self, invoices_data):
         moves = self.env['account.move']
         for move, move_data in invoices_data.items():
             if 'it_edi_send' in move_data['extra_edis']:
-                moves |= move
                 if attachment := move.l10n_it_edi_attachment_id:
                     attachments_vals[move] = {'name': attachment.name, 'raw': attachment.raw}
-                else:
+                    moves |= move
+                elif move_data.get('l10n_it_edi_values'):
                     attachments_vals[move] = invoices_data[move]['l10n_it_edi_values']
```

Comment:
```suggestion
                elif edi_values := move_data.get('l10n_it_edi_values'):
                    attachments_vals[move] = edi_values
```
Just to be consistent with the syntax of the previous statement :innocent: